src/main/java/com/auth0/net/BaseRequest.java

```diff
@@ -20,7 +22,7 @@
 
     protected abstract okhttp3.Request createRequest() throws Auth0Exception;
 
-    protected abstract T parseResponse(Response response) throws Auth0Exception;
+    protected abstract T parseResponseBody(okhttp3.Response response) throws Auth0Exception;
```

Comment:
Just thinking out loud, should we use something like `Auth0Request`, `Auth0Response` to avoid namespace clash? `Response#getBody` feels too generic that it will be used by everyone.

Reply:
I'd be ok with that. It's what I had originally, but typing out `Auth0XXX` for each type seemed excessive, and we already have `com.auth0.net.Request`.

Reply:
The repetition isn't that big a deal, and autocomplete can do the heavy lifting anyways. Some libraries basically namespace their types within the class name, others (like the Azure java SDK) just rely on the package name to provide the namespace.

I wouldn't want to have `Request` (existing type name) and `Auth0Response` though. So we'd need to rename the existing `Request` type to be `Auth0Request`, which I'm not sure is worth it. And if we do that, it should be in a separate PR.